index.html

```diff
@@ -208,91 +285,136 @@ <h2>
-            <p>
-              This ensures a user can leave a document that constantly attempts
-              to lock the pointer. The document will be blocked from locking
-              upon initial navigation or re-acquiring lock unless the user
-              re-engages directly with the document.
-            </p>
-            <p>
-              Conversely, if pointer lock is exited via {{exitPointerLock()}}
-              no <a>engagement gesture</a> is required to reenter pointer lock.
-              This enables applications that frequently move between
-              interaction modes, and ones that may do so based on a timer or
-              remote network activity.
-            </p>
-            <p data-link-for="Element">
-              Pointer lock is commonly combined with fullscreen [[FULLSCREEN]],
-              and if an <a>engagement gesture</a> is used to enter fullscreen
-              it is sufficient for a subsequent {{requestPointerLock()}}.
-            </p>
-          </aside>
-          <p>
-            If any element (including this one), whose <a>shadow-including
-            root</a> is same as this element's <a>shadow-including root</a>, is
-            already locked (or pending lock) the pointer lock <a>target</a>
-            must be updated to this element and a <a>pointerlockchange</a>
-            event sent.
-          </p>
-          <p>
-            If any element, whose <a>shadow-including root</a> is a different
-            document, is already locked the request must fail and a
-            <a>pointerlockerror</a> event be sent.
-          </p>
-          <p>
-            Once in the locked state the <a>user agent</a> must fire all
-            relevant user generated {{MouseEvent}} events (for example:
-            `mousemove`, `mousedown`, `mouseup`, `click`, and `wheel`)
-            [[ui-events]] to the <a>target</a> of pointer lock, and not fire
-            mouse events to other elements. Events that require the concept of
-            a mouse cursor must not be dispatched (for example: `mouseover`,
-            `mouseout`, `drag`, and `drop`).
-          </p>
-          <p>
-            In the locked state the system mouse cursor must be hidden.
-            Movement and button presses of the mouse must not cause the window
-            to lose focus.
-          </p>
-          <p>
-            Synthetic mouse events created by application script act the same
-            regardless of lock state.
+            RequestPointerLock() will follow these steps
+            <ol>
+              <li>
+                If the the <a>target</a>'s
+                <a>shadow-including root</a> is the <a>active document</a> of a
+                [=Document/browsing context=] which is (or has an <a>ancestor
+                browsing context</a> which is) in focus by a window which is in
+                focus by the operating system's window manager queue <a>pointerlockerror</a> 
```

Comment:
So, for all these error cases, the following is the more precise phrasing:

1. [Queue an element task](https://html.spec.whatwg.org/multipage/webappapis.html#queue-an-element-task) on the ??? task source, given [this](https://heycam.github.io/webidl/#this), to perform the following steps:
   1. [Fire an event](https://dom.spec.whatwg.org/#concept-event-fire) named `pointerlockerror` at  [this](https://heycam.github.io/webidl/#this)'s [node document](https://dom.spec.whatwg.org/#concept-node-document).
    1. Return [a promise rejected with](https://heycam.github.io/webidl/#a-promise-rejected-with) a "[`WrongDocumentError`](https://heycam.github.io/webidl/#wrongdocumenterror)" [`DOMException`](https://heycam.github.io/webidl/#idl-DOMException).

As for the task source, it may be worth checking implementations (e.g. just inspecting the Chromium source) to see what task source is in use. It might be a dedicated new one, in which case somewhere in the document you'll want to define the **pointer lock task source** and reference that. ([Example of another spec](https://wicg.github.io/portals/#portal-task-source).)